akka-stream-tests/src/test/scala/akka/stream/io/OutputStreamSourceSpec.scala

```diff
@@ -204,5 +209,27 @@ class OutputStreamSourceSpec extends StreamSpec(UnboundedMailboxConfig) {
 
       assertNoBlockedThreads()
     }
+
+    "correctly complete the stage after close" in assertAllStagesStopped {
+      // actually this was a race, so it only happened in at least one of 20 runs
+
+      val bufSize = 4
+      val sourceProbe = TestProbe()
+      val (outputStream, probe) = StreamConverters.asOutputStream(timeout)
+        .addAttributes(Attributes.inputBuffer(bufSize, bufSize))
+        .toMat(TestSink.probe[ByteString])(Keep.both).run
+
+      // fill the buffer up
+      (1 to (bufSize - 1)).foreach(outputStream.write)
+      Future {
+        outputStream.close()
+      }
+      // here is the race, has the elements reached the stage buffer yet?
+      Thread.sleep(500)
```

Comment:
this is ofc also racy, but adding the job to the queue should be relatively fast everywhere, this consistently failed on my machine, and now passes with the fix.